crates/iceberg/src/transaction.rs

```diff
@@ -95,6 +104,42 @@ impl<'a> Transaction<'a> {
         Ok(self)
     }
 
+    /// Creates a fast append action.
+    pub fn fast_append(
+        self,
+        commit_uuid: Option<String>,
+        key_metadata: Vec<u8>,
+    ) -> Result<FastAppendAction<'a>> {
+        let parent_snapshot_id = self
+            .table
+            .metadata()
+            .current_snapshot()
+            .map(|s| s.snapshot_id());
+        let snapshot_id = parent_snapshot_id.map(|id| id + 1).unwrap_or(0);
```

Comment:
The snapshot ID is a random int, and it should be checked if it hasn't been used before:

https://github.com/apache/iceberg-python/blob/f72e363b18baa181c998bbdef657982159a22d48/pyiceberg/table/metadata.py#L315-L326